config-ui/src/components/Nav.jsx

```diff
@@ -26,11 +26,41 @@ import {
 import '@/styles/nav.scss'
 import { ReactComponent as SlackIcon } from '@/images/slack-mark-monochrome-black.svg'
 import { ReactComponent as SlackLogo } from '@/images/slack-rgb.svg'
+import UIContext from '../store/UIContext'
+import useWindowSize from '../hooks/useWIndowSize'
+
 
 const Nav = () => {
+  const uiContext = useContext(UIContext);
+  const [menuClass, setMenuClass] = useState("navbarMenuButton");
+  const size = useWindowSize();
+
+  const toggleSidebarOpen = (open) => {
+    uiContext.changeSidebarVisibility(open)
+    if (open){
+      setMenuClass("navbarMenuButtonSidebarOpened");
+      return;
+    }
+    setMenuClass("navbarMenuButton");
+  }
+
+  useEffect(() => {
```

Comment:
This Effect could also be refactored to be more declarative and remove the nested `IF` statements
```suggestion
  useEffect(() => {
        toggleSidebarOpen(size.width >= 850 && uiContext.sidebarVisible != true ? true : false)
  }, [size]);
```